pyfar/classes/audio.py

```diff
@@ -1301,6 +1492,36 @@ def _power(a, b):
     return a**b
 
 
+def _matrix_multiplication(a, b, axes, dtypes):
+    # copy axes because it could be used multiple times if matrix
+    # multiplication is called with a tuple containing more than three elements
+    axes = axes.copy()
+
+    # adjust axes according to time/frequency dimension
+
+    #  handle first input
+    if dtypes[0] in (Signal, TimeData, FrequencyData):
+        a = np.expand_dims(a, 0) if a.ndim == 2 else a
+        axes[0] = tuple([ax-1 if ax < 0 else ax for ax in axes[0]])
+    elif a.ndim < b.ndim:
+        a = np.expand_dims(a, -1)
+        axes[0] = tuple([ax-1 if ax < 0 else ax for ax in axes[0]])
```

Comment:
Is this required? An axis is already appended in `_get_arithmetic_data` (line 1466) if the input is an array and any input is a pyfar audio object. If all inputs are are arry likes, we probably shouldn't do any processing.

If it is not required, could the function could be simplified?
```python
#  handle first input
    if dtypes[0] in (Signal, TimeData, FrequencyData):
        a = np.expand_dims(a, 0) if a.ndim == 2 else a

    # handle second input
    if dtypes[1] in (Signal, TimeData, FrequencyData):
        b = np.expand_dims(b, 1) if b.ndim == 2 else b

    # adjust axis for output if output is a pyfar audio object
    if any(dt in (Signal, TimeData, FrequencyData) for dt in dtypes):
        axes[0] = tuple([ax-1 if ax < 0 else ax for ax in axes[0]])
        axes[1] = tuple([ax-1 if ax < 0 else ax for ax in axes[1]])
        axes[2] = tuple([ax-1 if ax < 0 else ax for ax in axes[2]])
```

Reply:
Thanks for noting the appending in ``_get_arithmetic_data``. It was possible to simplify the code by using the ``audio_type`` variable in ``_get_arithmetic_data`` and ``_matrix_multiplication``.